src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -19,7 +20,35 @@ namespace System.Diagnostics
     /// </devdoc>
     public class DefaultTraceListener : TraceListener
     {
-        private const int InternalWriteSize = 16384;
+        private class DefaultTraceDebugProvider : DebugProvider
+        {
+            private const int InternalWriteSize = 16384;
+            private static readonly Lazy<DebugProvider> lazy = new Lazy<DebugProvider>(() => new DefaultTraceDebugProvider());
```

Comment:
You do not need `Lazy` here. Static property works just fine and it costs fraction of what the `Lazy` pattern costs.